soda_machine.rb

```diff
@@ -1,18 +1,35 @@
+require_relative "soda.rb"
+
 class SodaMachine
   attr_reader :sodas, :cash
+  @@inventory = 0
 
   def initialize(args = {})
     @sodas = args[:sodas]
+    self.class.add_soda(@sodas.length)
     @cash = args[:cash]
   end
 
-  def current_inventory_count
+  def self.current_inventory_count
+    return @@inventory
   end
 
   def find_soda(soda_brand)
+    @sodas.each{|x| return x.brand if x.brand == soda_brand}
```

Comment:
Close.  What kind of thing does the test want you to return?  What are you returning?

Also, what if you don't find it?  What are you returning then?  What does it want?